### ISSUE TITLE
Upgrade test for knative-eventing-kafka

### ISSUE THREAD

**Issue by sayanh** [kyma-project/kyma#8872]
**Timebox: 2days**

**Description**

The current upgrade scenario doesn't cover eventing implementation using knative-eventing-kafka. The goal here is to automate the upgrade test using knative-eventing-kafka.

**Reasons**

Automation

**AC**
- Lets Create a Gardener Azure Job to test upgrade scenario for knative-eventing-kafka
- Job runs as pre-submit jobs on PR similar to *pre-master-kyma-integration-gardener-azure*

**Comment by nachtmaar** [comment]
Seems there is a periodic gardener upgrade job: https://github.com/kyma-project/test-infra/pull/2419 which has been introduced recently.

**Comment by nachtmaar** [comment]
Instructions for poor mans upgrade:
```
$ kyma-knative-kafka-gardener-azure-provision-install <cluster-name> <old-kyma-version> # make sure to use matching kyma cli version and local kyma code checked out
$ kyma-appnpair <cluster-domain>
$ kyma-knative-kafka-gardener-azure-provision-install <cluster-name> <new-kyma-version> # make sure to use matching kyma cli version and local kyma code checked out
```

- Then check if events are still being received in the lambda function:
```bash
$ while true; do kyma-send-ce-structured km13upg2.kymatunas.shoot.canary.k8s-hana.ondemand.com; sleep 1; done
```
- Then verify tracing/metrics

**Comment by grischperl** [comment]
Addressed by [#2936](https://github.com/kyma-project/test-infra/pull/2936)
